tests/foreman/cli/test_remoteexecution.py

```diff
@@ -367,3 +424,137 @@ def test_positive_run_scheduled_job_template(self):
                     )
                 )
             )
+
+    @tier3
+    def test_positive_run_default_job_template_multiple_hosts(self):
+        """Run default job template against multiple hosts
+
+        :id: 415c0156-be77-4676-918b-c0d4be810b0e
+
+        :expectedresults: Verify the job was successfully ran against all hosts
+        """
+        with VirtualMachine(distro=DISTRO_RHEL7) as client2:
+            client2.install_katello_ca()
+            client2.register_contenthost(
+                    self.org['label'], lce='Library')
+            add_remote_execution_ssh_key(client2.ip_addr)
+            invocation_command = make_job_invocation({
+                'job-template': 'Run Command - SSH Default',
+                'inputs': 'command="ls"',
+                'search-query': "name ~ {0} or name ~ {1}".format(
+                    self.client.hostname, client2.hostname),
+            })
+            # collect output messages from clients
+            output_msgs = []
+            for vm in self.client, client2:
+                output_msgs.append('host output from {0}: {1}'.format(
+                        vm.hostname,
+                        ' '.join(JobInvocation.get_output({
+                            'id': invocation_command[u'id'],
+                            'host': vm.hostname})
+                        )
+                    )
+                )
+            self.assertEqual(invocation_command['success'], u'1', output_msgs)
+
+    @tier3
+    def test_positive_install_multiple_packages_with_a_job(self):
+        """Run job to install several packages on host
+
+        :id: 1cf2709e-e6cd-46c9-a7b7-c2e542c0e943
+
+        :expectedresults: Verify the packages were successfully installed
+            on host
+        """
+        # Create a custom repo
+        setup_org_for_a_custom_repo({
+            u'url': FAKE_0_YUM_REPO,
+            u'organization-id': self.org['id'],
+            u'content-view-id': self.content_view['id'],
+            u'lifecycle-environment-id': self.env['id'],
+            u'activationkey-id': self.activation_key['id'],
+        })
+        invocation_command = make_job_invocation({
+            'job-template': 'Install Package - Katello SSH Default',
+            'inputs': 'package={0} {1} {2}'.format(
+                "cow", "dog", "lion"),
+            'search-query': "name ~ {0}".format(self.client.hostname),
+        })
+        self.assertEqual(
+                invocation_command['success'],
+                u'1',
+                'host output: {0}'.format(
+                    ' '.join(JobInvocation.get_output({
+                        'id': invocation_command[u'id'],
+                        'host': self.client.hostname})
+                    )
+                )
+            )
+        result = ssh.command(
+                "rpm -q {0}".format("cow dog lion"),
```

Comment:
perhaps define a nice list  at the beginning of the test for putting the package names into it and use its reference everywhere you need. This way you'll get a single reference point in case you need to change the package names or their count and the string substitution will be dynamic, so it won't require additional tweaking (like adding more '{x}-es'

```python
packages = ['cow', 'dog', 'lion']
ssh.command("rpm -q {0}".format(' '.join(packages))
```